### DISCUSSION TITLE
Set combination setter number

### DISCUSSION THREAD

**Discussion by bv74** [GrandOrgue/grandorgue#1679] [Help] [answered]
HI, 

Linked with https://github.com/GrandOrgue/grandorgue/issues/1237, I understand that we can't set GUI combination number with a MIDI event. Is that right ? 
So we can't yet choose a specific combination with a keyboard (only figures an Enter button) from the organ ? 
Thank you for your help. 
BV

**Comment by larspalo** [comment, marked as answer]
You can get to a specific setter combination from the toolbar of GrandOrgue (enter a value by keyboard). There are many ways to navigate the setter combinations by MIDI messages, but not (yet) a direct number input way except for within each "bank" of 10s (where you can use the __0 to __9 buttons).

From the Combination Setter panel you can assign the different + and - values buttons as well as the Current button to different MIDI buttons to navigate to any combination in a few steps. To get to combination number 249 from combination 008 you can for instance, in a sequence push:

`+100, +100, +10, +10, +10, +10, +1, Current`

or

`+100, +100, +10, +10, +10, +10, Next`

or

`+100, +100, +10, +10, +10, +10, __9`

The number list wraps around so to get to 998 from 008 you can just push `-10, Current`.

This is obviously something you only do between pieces you play, not while playing! Anyone that thinks about direct number input while playing has not understood how the combination (sequencer) setter is supposed to be used - I'm not only talking about within GrandOrgue but very much IRL situations too. (You're really supposed to program the combinations for a piece in sequence so that you only ever need to push the Next button to advance to next combination while playing) For a specific performance it's common to create a completely new list of combinations so that for the whole concert you only step forward. You can also create a new list for each piece you play and just load that combination list for every piece you're about to play.

For liturgical playing or impromptu improvisation you're actually better off using Divisionals and/or Generals if you want to jump between combinations "randomly".

**Reply by bv74** [reply]
Thank you very much for your answer. 
My question was only concerning setting before playing and not during playing, for sure, to reach more quickly the right combination for the piece avoiding moving stops during this preparation.
